crates/matrix-sdk/src/sliding_sync/list/mod.rs

```diff
@@ -1524,8 +1220,12 @@ mod tests {
         let (sender, _receiver) = channel(4);
 
         let mut list = SlidingSyncList::builder("testing")
-            .sync_mode(SlidingSyncMode::new_selective())
-            .ranges(vec![0..=10, 42..=153].to_vec())
+            .sync_mode(
```

Comment:
Let's rename the test to `test_generator_changing_sync_mode_to_various_modes`. Other tests are all changing the sync mode, but only onto the same mode. This one is changing the sync mode to various modes.